ui/src/flux/components/FluxNew.tsx

```diff
@@ -2,14 +2,26 @@ import React, {PureComponent, ChangeEvent, FormEvent} from 'react'
 
 import FluxForm from 'src/flux/components/FluxForm'
 
-import {NewService, Source, Notification} from 'src/types'
-import {fluxCreated, fluxNotCreated} from 'src/shared/copy/notifications'
-import {CreateServiceAsync} from 'src/shared/actions/services'
+import {NewService, Source, Service, Notification} from 'src/types'
+import {
+  fluxCreated,
+  fluxNotCreated,
+  notifyFluxNameAlreadyTaken,
+} from 'src/shared/copy/notifications'
+import {
+  CreateServiceAsync,
+  SetActiveServiceAsync,
+} from 'src/shared/actions/services'
+import {FluxFormMode} from 'src/flux/constants/connection'
+import {getDeep} from 'src/utils/wrappers'
 
 interface Props {
   source: Source
-  onDismiss: () => void
+  services: Service[]
+  setActiveFlux?: SetActiveServiceAsync
+  onDismiss?: () => void
   createService: CreateServiceAsync
+  router?: {push: (url: string) => void}
```

Comment:
what does this braces syntax mean?

Reply:
its an object that  has a function push on it.